mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulEditProfileNameDialog.kt

```diff
@@ -0,0 +1,75 @@
+package ch.epfl.sdp.mobile.state
+
+import androidx.compose.foundation.MutatePriority
+import androidx.compose.foundation.MutatorMutex
+import androidx.compose.runtime.*
+import ch.epfl.sdp.mobile.application.authentication.AuthenticatedUser
+import ch.epfl.sdp.mobile.ui.setting.EditProfileNameDialog
+import ch.epfl.sdp.mobile.ui.setting.EditProfileNameDialogState
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+/**
+ * Implementation of the [EditProfileNameDialogState] interface
+ *
+ * @param user the current [AuthenticatedUser].
+ * @param scope the coroutine scope.
+ * @param onSave the callback called after we saved the changes.
+ * @param onCancel the callback called when we click the cancel button.
```

Comment:
Parameter names should be updated.
```suggestion
 * @param onSaveAction the callback called after we saved the changes.
 * @param onCancelAction the callback called when we click the cancel button.
```